### PR TITLE
Add Google Drive Sync and sign in support.

### PR DESCRIPTION
**Pull request by vannem-sj**

### CODE REVIEW

File: OpenScienceJournal/app/src/main/java/com/google/android/apps/forscience/whistlepunk/opensource/accounts/GoogleAccountsProvider.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  @Override
+  public int getAccountCount() {
+    GoogleSignInAccount account = GoogleSignIn.getLastSignedInAccount(context);
+    return account == null ? 0 : 1;
+  }
+
+  @Override
+  public void installAccountSwitcher(ActivityWithNavigationView activity) {
+
+  }
```
Comment (ashley3) [inline OpenScienceJournal/app/src/main/java/com/google/android/apps/forscience/whistlepunk/opensource/accounts/GoogleAccountsProvider.java]: Liz probably knows this code better than me but does this being blank mean we can only sign into one account and not switch between them? If so, is that infeasible or should we file a bug to address that in the future?

Comment (vannem-sj) [reply OpenScienceJournal/app/src/main/java/com/google/android/apps/forscience/whistlepunk/opensource/accounts/GoogleAccountsProvider.java]: Correct. We can't switch between accounts. You sign in, then sign out, then sign in as someone else, if you want.

However, this method is actually the part of the code where we added the component to our app. We no longer do that at all, just intent off to a system dialog. So I don't think we'd have to do anything here anyway.

**Comment by vannem-sj** [comment]
> I might just be missing it in the preview on my laptop but it looks like ic_account_circle_black might be missing for mdpi and xxhdpi?

Oh, yeah. Whoops. I assumed they'd just fail to the other versions. I'll add them.

**Comment by ashley3** [comment]
LGTM

**Pull request merged**